### ISSUE TITLE
Picking `startBlock` for substreams data source

### ISSUE THREAD

**Issue by YaroShkvorets** [graphprotocol/graph-node#4449]
[This logic](https://github.com/graphprotocol/graph-node/blob/860becc40e9d4350adc1372e430f354c3ae18727/chain/substreams/src/data_source.rs#L175-L178) of picking the startBlock for the module to stream, where we pick the MIN `initial_block` among **all** modules in the package, seems superfluous and wrong.

For some complex substreams it could pick wrong block values resulting in indexing errors like this:
```
An error occurred while streaming blocks: status: InvalidArgument, message: "start block 100000 smaller than request outputs for module \"graph_out\" with start block 1000000"
```

IMO We should just take the `initial_block` of the module being requested.

`substreams pack` already assigns initial_block for modules that don't have `initial_block` set, based on its ancestors, so trying to silently override it here seems unnecessary.

**Comment by maoueh** [comment]
Yeah you are right it's wrong in some circumstances indeed, this logic was taken from how `subgraph` deals with multi start block in manifest, but it cannot apply the same to `Substreams`.

The right solution would actually I think let "substreams" determine it, because it requires the exact module "graph" that is going to be needed.

But yeah an easy fix right now is to do what you said, just take the actual module. Since we only accept `.spkg` input AFAIK, we are good. If we move to support also from `substreams.yaml` directly, we might need to revisit.
